src/saiga/core/geometry/sphere.cpp

```diff
@@ -68,6 +68,15 @@ float Sphere::sdf(vec3 p) const
     return (p - pos).norm() - r;
 }
 
+vec2 Sphere::projectedIntervall(const vec3& d) const
+{
+    vec2 ret;
+    float t = dot(d, pos);
+    ret[0]  = std::min(t - r, t + r);
+    ret[1]  = std::max(t + r, t - r);
```

Comment:
if (r >= 0) the min/max is not required. Better to remove min/max and add a SAIGA_ASSERT(r >=0);